analyzers/DesyncSafeAnalyzer.Tools/UnsyncUtils.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using static War3Api.Common;
+
+namespace DesyncSafeAnalyzer.Tools
+{
+  /// <summary>
+  /// Provides methods that make it easier to safely call unsynchronized code.
+  /// </summary>
+  public static class UnsyncUtils
+  {
+    /// <summary>
+    /// Executes the specified function for a specific client.
+    /// </summary>
+    /// <param name="functionToExecute">The function to be executed.</param>
+    /// <param name="whichPlayer">The client to execute the function for.</param>
+    public static void InvokeForClient(Action functionToExecute, player whichPlayer)
+    {
+      if (whichPlayer != null && GetLocalPlayer() == whichPlayer)
+        functionToExecute();
+    }
+
+    /// <summary>
+    /// Invokes the specified function for clients matching a provided condition.
+    /// </summary>
+    /// <param name="functionToExecute">The function to be executed.</param>
+    /// <param name="condition">The condition a client must pass for the function to be executed.</param>
+    public static void InvokeForClients(Action functionToExecute, Func<player, bool> condition)
```

Comment:
This has to be removed entirely. It's way too easy to produce unsafe code through the condition parameter.